viewing_party/main.py

```diff
@@ -0,0 +1,221 @@
+
+def create_movie(title, genre, rating): 
+    movie = dict();
+    if title:
+        movie['title'] = title;
+    else:
+        return None
+    if genre:
+        movie['genre'] = genre;
+    else:
+        return None
+    if rating:
+        movie['rating'] = rating;
+    else:
+        return None
+    return movie
+
+
+def add_to_watched(user_data, movie):   
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    watched_movie_temp_list = []
+
+    for key, values in user_data.items():
+        for movie_info in values:
+            if title == movie_info["title"]:
+                user_data["watchlist"].remove(movie_info)
+                watched_movie_temp_list.append(movie_info)
+    for watched_movie in watched_movie_temp_list:
+        user_data["watched"].append(watched_movie)
+
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+
+    average_ratings = []
+    average = []
+
+    if not user_data["watched"]:
+        return 0.0
+
+    for keys, values in user_data.items():
+        for watched_movies in values:   
+            if watched_movies["rating"] == '':
+                watched_movies["rating"] = 0.0
+            else:
+                average_ratings.append(float(watched_movies["rating"]))
+                average = sum(average_ratings)/len(average_ratings)
+        return average
+
+def get_most_watched_genre(user_data):
+
+    genre_count = []
+    genre_stats = dict()
+    most_watched_genre = ""
+
+    if not user_data["watched"]:
+        return None 
+    for keys, values in user_data.items():
+        for watched_movies in values:
+            genre_count.append(watched_movies['genre'])
+    
+
+    for value in genre_count:
+        if value not in genre_stats:
+            genre_stats[value] = 1
+        else:
+            genre_stats[value] +=1
+
+    most_watched_genre_tuple = [(value, key) for key, value in genre_stats.items()]
+    most_watched_genre = max(most_watched_genre_tuple)[1]
+    return most_watched_genre
+
+
+def get_unique_watched(user_data):
+    user_watched = []
+    friends_watched = []
+    movies_unique_to_friends_list = []
+    
+
+    for element in user_data["friends"]:
+        for k,v in element.items():
+            for item in v:
+                for k,v in item.items():
+                    friends_watched.append(v)
+    
+    for element in user_data["watched"]:
+        for k,v in element.items():
+            user_watched.append(v)
+            if v not in friends_watched:
+                movies_unique_to_friends_list.append(element)  
```

Comment:
If you're using generic names like `k`, `v` it can make your code tricky to read but also can be a sign that you've got an unnecessary loop: 

```suggestion
    for friend in user_data["friends"]:
        for movie in friend["watched"]:
            friends_watched.append(movie["title"])
    
    for movie in user_data["watched"]:
        user_watched.append(movie["title"])
        if movie["title"] not in friends_watched:
            movies_unique_to_friends_list.append(movie)  
```